### PR TITLE
Move relThField to ini

### PR DESCRIPTION
**Pull request by awirb**
* address issue #627 
* address issue #580 

TODO:
* what if massPerParticleDeterminationMethod = nppk
* -> also prob for simiSol

### CODE REVIEW

**Review by matthiasto — APPROVED** [review]

File: avaframe/ana1Tests/simiSolTest.py [hunk 1]
```diff
@@ -737,7 +738,8 @@ def analyzeResults(particlesList, fieldsList, solSimi, fieldHeader, cfgSimi, out
             LMax error on flow velocity for saved time steps
 
     """
-    relTh = simDFrow['relTh']
+    # relTh = simDFrow['relTh']
```
Comment (fso42) [inline avaframe/ana1Tests/simiSolTest.py:741]: remove?

File: avaframe/com1DFA/com1DFA.py [hunk 2]
```diff
@@ -462,10 +461,23 @@ def prepareInputData(inputSimFiles, cfg):
...
 
+    # read data from relThFile
+    if relThFile != '':
+        relThField = IOf.readRaster(relThFile)
+        relThFieldData = relThField['rasterData']
+        if demOri['header']['ncols'] != relThField['header']['ncols'] or demOri['header']['ncols'] != relThField['header']['ncols']:
+            message = ('Release thickness field read from %s does not match the number of rows \
+                       and columns of the dem' % inputSimFiles['relThFile'])
+            log.error(message)
+            raise AssertionError(message)
```
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:477]: I think this is an important error to check for via pytest!

File: avaframe/com1DFA/com1DFA.py [hunk 3]
```diff
@@ -881,7 +892,14 @@ def initializeParticles(cfg, releaseLine, dem, inputSimLines='', logName='', rel
     areaRaster = dem['areaRaster']
 
     # get the initialization method used
-    massPerPart, nPPK = com1DFATools. getPartInitMethod(cfg, csz)
+    if len(relThField) != 0:
+        relThForPart = np.amax(relThField)
+    elif cfg.getboolean('relThFromShp'):
+        relThForPart = np.amax(np.asarray(releaseLine['thickness'], dtype=float))
```
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:898]: This should be tested

File: avaframe/com1DFA/deriveParameterSet.py [hunk 4]
```diff
@@ -400,6 +389,7 @@ def setThicknessValueFromVariation(key, cfg, simType, row):
             cfg['GENERAL'][key] = variationIni
 
         else:
+            print('VAR', variationFactor, cfg['GENERAL'][thType],  str(float(cfg['GENERAL'][thType]) * variationFactor))
```
Comment (fso42) [inline avaframe/com1DFA/deriveParameterSet.py:392]: needed? then move to log

File: avaframe/in1Data/getInput.py [hunk 5]
```diff
@@ -228,6 +248,8 @@ def getAndCheckInputFiles(inputDir, folder, inputType):
         subfolder name where the shape file should be located (SECREL, ENT or RES)
     inputType : str
         type of input (used for the logging messages). Secondary release or Entrainment or Resistance
+    fileExt: str
+        file extension e.g. shp, asc - optional default is shp
```
Comment (fso42) [inline avaframe/in1Data/getInput.py:252]: At the moment it is not set to a default value. If nothing is provided fileext is none. 
I think it should be optional as mentioned therefore line 238 needs fileExt='shp'
Furthermore: all `shp` infos above (when calling this function) can be removed

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/com1DFA/com1DFA.py (hunk 2 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:477]: included in pytest

File: avaframe/com1DFA/com1DFA.py (hunk 3 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:898]: moved this to a separate function and included a pytest

File: avaframe/com1DFA/deriveParameterSet.py (hunk 4 above)
Comment (awirb) [reply avaframe/com1DFA/deriveParameterSet.py:392]: removed

File: avaframe/ana1Tests/simiSolTest.py (hunk 1 above)
Comment (awirb) [reply avaframe/ana1Tests/simiSolTest.py:741]: I left this as simiSol needs more updates - there is a PR on this

File: avaframe/in1Data/getInput.py (hunk 5 above)
Comment (awirb) [reply avaframe/in1Data/getInput.py:252]: made it optioanl and default to shp

**Review by fso42 — APPROVED** [review]

**Pull request merged**